carp.protocols.core/src/commonMain/kotlin/dk/cachet/carp/protocols/domain/StudyProtocol.kt

```diff
@@ -127,7 +127,9 @@ class StudyProtocol private constructor( val ownerId: UUID, val name: String, va
      * Add a [masterDevice] which is responsible for aggregating and synchronizing incoming data.
      * Its role name should be unique in the protocol.
      *
-     * @throws IllegalArgumentException in case a device with the specified role name already exists.
+     * @throws IllegalArgumentException when:
+     *  - a device with the specified role name already exists
```

Comment:
Why is it relevant?

Reply:
Device role name being unique? It's the unique key that determines the _source_ of the data. It's also [part of the identity of a "data stream"](https://github.com/cph-cachet/carp.core-kotlin/blob/develop/carp.data.core/src/commonMain/kotlin/dk/cachet/carp/data/application/DataStreamId.kt#L13).

If it wouldn't be unique, e.g., you'd name two SENS patches "SENS patch", how would you know which one is the one collecting data for the right leg and which one is the one collecting data on the left leg? And, the data would become the same "data stream", thus data from both sensors would be bunched together and you could no longer tell them apart.